napari/utils/events/containers/_evented_dict.py

```diff
@@ -0,0 +1,108 @@
+"""MutableMapping that emits events when altered."""
+from typing import Mapping, Sequence, Type, Union
+
+from ..event import EmitterGroup, Event
+from ..types import SupportsEvents
+from ._dict import _K, _T, TypedMutableMapping
+
+
+class EventedDict(TypedMutableMapping[_K, _T]):
+    """Mutable dictionary that emits events when altered.
+
+    This class is designed to behave exactly like builting ``dict``, but
+    will emit events before and after all mutations (addition, removal, and
+    changing).
+
+    Parameters
+    ----------
+    data : Mapping, optional
+        Dictionary to initialize the class with.
+    basetype : type of sequence of types, optional
+        Type of the element in the dictionary.
+
+    Events
+    ------
+    changed (key: K, old_value: T, value: T)
+        emitted when ``key`` is set from ``old_value`` to ``value``
+    adding (key: K)
+        emitted before an item is added to the dictionary with ``key``
+    added (key: K, value: T)
+        emitted after ``value`` was added to the dictionary with ``key``
+    removing (key: K)
+        emitted before ``key`` is removed from the dictionary
+    removed (key: K, value: T)
+        emitted after ``key`` was removed from the dictionary
+    refresh (key, K, value: T)
+        emitted after ``value`` of ``key`` was changed. Only implemented by subclasses to give them an option
+        to trigger some update after ``value`` was changed and this class did not register it.
```

Comment:
is 'refresh' a standard event name in either Qt or napari? I feel like `updated` better matches the description

Reply:
Hm, you are right, `refresh` is not really used throughout qt or napari. I am happy to change to `updated` instead. Nice catch!